src/drawline.c

```diff
@@ -4116,7 +4116,7 @@ win_line(
 		    || text_prop_next <= last_textprop_text_idx
 #endif
 		    || (wp->w_p_list && wp->w_lcs_chars.eol != NUL
-						&& wlv.p_extra != at_end_str)
+						&& lcs_eol_one != -1)
```

Comment:
This makes the `at_end_str` variable no longer necessary, as it is now only used once in `wlv.p_extra = at_end_str`, which can now be changed to just `wlv.p_extra = ""`.